lib/stencil-bundle.js

```diff
@@ -151,51 +152,98 @@ class Bundle {
+                ]),
+            );
+        }
 
-                    return async.parallel(
-                        [
-                            this._checkObjects.bind(this, results),
-                            this._detectCycles.bind(this, results),
-                        ],
-                        (checkingError) => {
-                            if (checkingError) {
-                                callback(checkingError);
-                            }
-
-                            console.log(`${'ok'.green} -- Template Parsing Finished`);
-                            callback(null, ret);
-                        },
-                    );
-                },
+        // eslint-disable-next-line node/no-unsupported-features/es-builtins
+        const res = await Promise.allSettled([
+            recursiveReadDir(this.templatesPath, ['!*.html']),
+            ...externalLibPaths,
+        ]);
+        const [{ value: internalTemplates }, ...externalTemplatesList] = res;
+        const internalPartials = internalTemplates.map((file) => {
+            return upath.toUnix(
+                file.replace(this.templatesPath + path.sep, '').replace(/\.html$/, ''),
             );
         });
+        const externalPartials = externalTemplatesList.reduce(
+            (partials, { value: externalTemplates }) => {
+                const extractedPartials = externalTemplates.map((file) => {
+                    const startposition = file.indexOf('node_modules');
+                    const endPosition = file.lastIndexOf('templates') + 'templates/'.length;
+                    const partialPrefix = file
+                        .slice(startposition, endPosition)
+                        .replace('node_modules', 'external');
+                    return upath.toUnix(
+                        partialPrefix +
+                            file
+                                .replace(this.externalTemplatesPath + path.sep, '')
+                                .replace(/\.html$/, ''),
+                    );
+                });
+                partials.push(...extractedPartials);
+
+                return partials;
+            },
+            [],
+        );
+        const allPartials = [...externalPartials, ...internalPartials];
+
+        let results;
+        try {
+            results = await async.map(
+                allPartials,
+                templateAssembler.assembleAndBundle.bind(null, this.templatesPath),
+            );
+
+            const ret = {};
+
+            allPartials.forEach((file, index) => {
+                ret[file] = results[index];
+            });
+
+            // eslint-disable-next-line no-unused-vars
+            const checkResult = await async.parallel([
+                this._checkObjects.bind(this, results),
+                this._detectCycles.bind(this, results),
+            ]);
+
+            console.log(`${'ok'.green} -- Template Parsing Finished`);
+
+            return ret;
```

Comment:
did you try to use `return callback(null, ret)` here? What happened?

Reply:
we received `task` object with field `templates` `undefined` which prevented us to create bundle